### ISSUE TITLE
about output layer

### ISSUE THREAD

**Issue by kelvinqin** [Svito-zar/gesticulator#12]
Dear Taras,
Today I revisited your code, a curiosity is about this line in model.py (line 329):
        **curr_pose = self.hidden_to_output(final_h) * math.pi #because it is from -pi to pi**

To me, multiplying math.pi is redundant because in your output layer, the final function is a linear transformation (line 161), which should be capable to learn that constant value automatically:
        self.hidden_to_output = nn.Sequential(nn.Linear(final_hid_l_sz, self.output_dim),
                                              nn.Tanh(), nn.Dropout(args.dropout),
                                              **nn.Linear(self.output_dim, self.output_dim)**)

In my case, I am working on 2D lip-synching, and both training input and ground-truth (I am using MFCC as input, face landmarks detected by dlib as ground truth) are already normalized by removing mean and divided by std, so I did not multiply anything with the output:
        curr_pose = self.hidden_to_output(final_h)

Not sure if I am doing it correctly?

Thanks for your sharing,
Kelvin

**Comment by Svito-zar** [comment]
Hi @kelvinqin, you are absolutely right, this part in the code is redundant and needs to be removed.
It was there before we added extra linear layer at the end.

And you are also right that there is no need for any multipliers, especially if your data has been normalized.

**Comment by kelvinqin** [comment]
Taras,
Thanks so much for your answer 👍  
Kelvin
